### ISSUE TITLE
libsmartcols: column flags does not apply after adding to table in JSON

### ISSUE THREAD

**Issue by ignatenkobrain** [karelzak/util-linux#254]
```c
#include <libsmartcols.h>

int main (void)
{
  struct libscols_table *tb;
  struct libscols_column *cl;
  struct libscols_line *ln;

  tb = scols_new_table ();
  cl = scols_table_new_column (tb, "TEST", 3, 0);
  scols_column_set_flags (cl, SCOLS_FL_TREE);

  ln = scols_table_new_line (tb, NULL);
  scols_line_set_data (ln, 0, "Foo");

  ln = scols_table_new_line (tb, ln);
  scols_line_set_data (ln, 0, "Bar");

  //scols_table_enable_json (tb, 1);

  scols_print_table (tb);
  scols_unref_table (tb);

  return 0;
}
```

In human-readable format it prints as:
```
TEST
Foo
`-Bar
```

But in JSON it prints as:
```
{
   "": [
      {"test": "Foo"},
      {"test": "Bar"}
   ]
}
```

But if I will move `SCOLS_FL_TREE` before attaching column to the table it will work.

**Comment by ignatenkobrain** [comment]
problem is that we're doing `tb->ntreecols++` in `scols_table_add_column()`, so we should change ntreecols on each `scols_column_set_flags()`, not when adding column.

So my idea when we are ataching column to table we will keep pointer to table in column and when we are going to change flags - if wasnt tree and now tree -> tb->ntreecols++ and vice-versa tb->ntreecols--;

@karelzak what do you think?

**Comment by ignatenkobrain** [comment]
or (seems easier) recount ntreecols after each `scols_column_set_flags()`

**Comment by ignatenkobrain** [comment]
Implemented thoughts in #255. @karelzak please review.

**Comment by karelzak** [comment]
Merged, thanks!
